doc: Add a new nRF52840 dongle "bare" variant (#23055)
nrfconnect/sdk-nrf: applications/nrf_desktop/board_configuration.rst

      * The board supports ``release`` :ref:`nrf_desktop_bluetooth_guide_fast_pair` configuration (``release_fast_pair`` file suffix).
        The configuration uses the MCUboot bootloader built in the direct-xip mode (``MCUBOOT+XIP``), and supports firmware updates using the :ref:`nrf_desktop_dfu` and the :ref:`nrf_desktop_dfu_mcumgr`.

> nRF52840 USB Dongle (``nrf52840dongle/nrf52840/bare``) and nRF52833 USB Dongle (``nrf52833dongle``)

Reviewer: Is this change only for the nRF desktop application? I could see the "nrf52840dongle/nrf52840" target used in other samples as well.
Also, I think we would require a changelog entry.

Author: Yes, this change is only for those, who disable nRF5 SDK bootloader. I was able to find this override only in nRF Desktop applications.

      * Since the nRF52840 Dongle is generic and defined in Zephyr, project-specific changes are applied in the DTS overlay file.
      * The application is configured to act as a dongle that forwards data from both mouse and keyboard.
      * Bluetooth uses Nordic Semiconductor's SoftDevice link layer and is configured to act as a central.
        Input data comes from Bluetooth and is retransmitted to USB.

>       * The configuration with the B0 bootloader is set as default for the ``nrf52840dongle/nrf52840/bare`` board and with the MCUboot bootloader is set as default for the ``nrf52833dongle`` board.

Reviewer 2: ```suggestion
      * For the ``nrf52840dongle/nrf52840/bare`` board, the configuration using the B0 bootloader is set as the default.
        For the ``nrf52833dongle`` board, the configuration using the MCUboot bootloader is set as the default.
```
